File: examples/java/documentation/src/main/java/com/mycompany/app/dataSources/DataSourcesRemoteState.java
package com.mycompany.app.dataSources;

import com.hashicorp.cdktf.TerraformStack;
import com.hashicorp.cdktf.NamedRemoteWorkspace;
import software.constructs.Construct;

import imports.aws.provider.AwsProvider;
import imports.aws.provider.AwsProviderConfig;

// DOCS_BLOCK_START:data-sources-remote-state-data-source
import com.hashicorp.cdktf.DataTerraformRemoteState;
import com.hashicorp.cdktf.DataTerraformRemoteStateRemoteConfig;
import imports.aws.instance.Instance;
import imports.aws.instance.InstanceConfig;

public class DataSourcesRemoteState extends TerraformStack {

    public DataSourcesRemoteState(Construct scope, String id) {
        super(scope, id);
        // DOCS_BLOCK_END:data-sources-remote-state-data-source
        AwsProvider provider = new AwsProvider(this, "provider", AwsProviderConfig.builder()
                .region("us-east-1")
                .build());
        // DOCS_BLOCK_START:data-sources-remote-state-data-source

        // ....
        DataTerraformRemoteState remoteState = new DataTerraformRemoteState(this, "state",
                DataTerraformRemoteStateRemoteConfig.builder()
                        .organization("hashicorp")
                        .workspaces(new NamedRemoteWorkspace("vpc-prod"))
                        .build());

        new Instance(this, "foo", InstanceConfig.builder()
                // ....
                .subnetId(remoteState.getString("subnet_id"))
                .build());
    }
}
// DOCS_BLOCK_END:data-sources-remote-state-data-source

File: examples/go/documentation/datasources-remote-state.go
// Copyright (c) HashiCorp, Inc.
// SPDX-License-Identifier: MPL-2.0

package main

// DOCS_BLOCK_START:remote-state-datasources
import (
	"github.com/aws/constructs-go/constructs/v10"
	"github.com/aws/jsii-runtime-go"
	"github.com/hashicorp/terraform-cdk-go/cdktf"
	"github.com/hashicorp/terraform-cdk/examples/go/documentation/generated/hashicorp/aws/instance"
	aws "github.com/hashicorp/terraform-cdk/examples/go/documentation/generated/hashicorp/aws/provider"
)

func RemoteStateDataSourceStack(scope constructs.Construct, name string) cdktf.TerraformStack {
	stack := cdktf.NewTerraformStack(scope, &name)

	aws.NewAwsProvider(stack, jsii.String("aws"), &aws.AwsProviderConfig{
		Region: jsii.String("eu-central-1"),
	})

	remote_state := cdktf.NewDataTerraformRemoteState(stack, jsii.String("remote_state"), &cdktf.DataTerraformRemoteStateRemoteConfig{
		Organization: jsii.String("hashicorp"),
		Workspaces:   cdktf.NewNamedRemoteWorkspace(jsii.String("vpc-prod")),
	})

	instance.NewInstance(stack, jsii.String("region"), &instance.InstanceConfig{
		SubnetId: remote_state.GetString(jsii.String("subnet_id")),
	})

	return stack
}

// DOCS_BLOCK_END:remote-state-datasources
